shapely/strtree.py

```diff
@@ -122,28 +129,59 @@ def __reduce__(self):
         else:
             return (STRtree, (self.geometries,))
 
-    def query_items(self, geom: BaseGeometry) -> Sequence[Any]:
-        """Query for nodes which intersect the geom's envelope to get
-        stored items.
+    def query_items(
+        self, geom: BaseGeometry, predicate=None, distance=None
+    ) -> Sequence[Any]:
+        """
+        Return the index of all geometries in the tree with extents that
+        intersect the envelope of the input geometry.
 
         Items are integers serving as identifiers for an application.
 
+        If predicate is provided, a prepared version of the input geometry
+        is tested using the predicate function against each item whose
+        extent intersects the envelope of the input geometry:
+        predicate(geometry, tree_geometry).
+
+        The 'dwithin' predicate requires GEOS >= 3.10.
+
+        If geometry is None, an empty array is returned.
```

Comment:
Returning empty result sequences is good in the way that Python slicing returns empty sequences. It designs errors out of existence.

Reply:
Yes, I think the same is true for an empty geometry (can add that)